products/storage/api.yaml

```diff
@@ -780,4 +780,60 @@ objects:
         name: timeUpdated
         api_name: updated
         description: The modification time of the object metadata.
+  - !ruby/object:Api::Resource
+    name: 'HmacKey'
+    kind: 'storage#hmacKey'
+    base_url: projects/{{project}}/hmacKeys
+    create_url: projects/{{project}}/hmacKeys?serviceAccountEmail={{serviceAccountEmail}}
+    self_link: projects/{{project}}/hmacKeys/{{accessId}}
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'hhttps://cloud.google.com/storage/docs/authentication/managing-hmackeys'
+      api: 'https://cloud.google.com/storage/docs/json_api/v1/projects/hmacKeys'
+    description: |
+      The hmacKeys resource represents an HMAC key within Cloud Storage. The resource 
+      consists of a secret and HMAC key metadata. HMAC keys can be used as credentials 
+      for service accounts. For more information, see HMAC Keys.
 
+      Note that the hmacKeys resource is only returned when you use hCREATE. Other methods, 
+      such as GET, return the metadata portion of the HMAC key resource.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'serviceAccountEmail'
+        description: The email address of the key's associated service account.
+        input: true
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'secret'
```

Comment:
It looks like the secret is only returned in the create response, and not during GET. As well, we'll need to store the access_id so we can retrieve it later. We have a custom code hook, `post_create` that will let you write both of those. This is a good example of a piece of existing custom code doing that (you'll likely want to set the id based on access_id):

https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/post_create/set_computed_name.erb

The secret should also be marked `ignore_read` so we don't overwrite it with nothing later on.